shell/platform/android/io/flutter/embedding/engine/systemchannels/PlatformChannel.java

```diff
@@ -4,26 +4,536 @@
+          requestedOrientation |= 0x02;
+          break;
+        case LANDSCAPE_RIGHT:
+          requestedOrientation |= 0x08;
+          break;
+      }
+
+      if (firstRequestedOrientation == 0x00) {
+        firstRequestedOrientation = requestedOrientation;
+      }
+    }
+
+    switch (requestedOrientation) {
+      case 0x00:
+        return ActivityInfo.SCREEN_ORIENTATION_UNSPECIFIED;
+      case 0x01:
+        return ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
+      case 0x02:
+        return ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE;
+      case 0x04:
+        return ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT;
+      case 0x05:
+        return ActivityInfo.SCREEN_ORIENTATION_USER_PORTRAIT;
+      case 0x08:
+        return ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE;
+      case 0x0a:
+        return ActivityInfo.SCREEN_ORIENTATION_USER_LANDSCAPE;
+      case 0x0b:
+        return ActivityInfo.SCREEN_ORIENTATION_USER;
+      case 0x0f:
+        return ActivityInfo.SCREEN_ORIENTATION_FULL_USER;
+      case 0x03: // portraitUp and landscapeLeft
+      case 0x06: // portraitDown and landscapeLeft
+      case 0x07: // portraitUp, portraitDown, and landscapeLeft
+      case 0x09: // portraitUp and landscapeRight
+      case 0x0c: // portraitDown and landscapeRight
+      case 0x0d: // portraitUp, portraitDown, and landscapeRight
+      case 0x0e: // portraitDown, landscapeLeft, and landscapeRight
+        // Android can't describe these cases, so just default to whatever the first
+        // specified value was.
+        switch (firstRequestedOrientation) {
+          case 0x01:
+            return ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
+          case 0x02:
+            return ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE;
+          case 0x04:
+            return ActivityInfo.SCREEN_ORIENTATION_REVERSE_PORTRAIT;
+          case 0x08:
+            return ActivityInfo.SCREEN_ORIENTATION_REVERSE_LANDSCAPE;
+        }
+    }
+
+    // Execution should never get this far, but if it does then we default
+    // to a portrait orientation.
+    return ActivityInfo.SCREEN_ORIENTATION_PORTRAIT;
   }
 
-  public void setMethodCallHandler(@Nullable MethodChannel.MethodCallHandler handler) {
-    channel.setMethodCallHandler(handler);
+  private AppSwitcherDescription decodeAppSwitcherDescription(@NonNull JSONObject encodedDescription) throws JSONException {
+    int color = encodedDescription.getInt("primaryColor");
+    if (color != 0) { // 0 means color isn't set, use system default
+      color = color | 0xFF000000; // color must be opaque if set
+    }
+    String label = encodedDescription.getString("label");
+    return new AppSwitcherDescription(color, label);
   }
 
+  private List<SystemUiOverlay> decodeSystemUiOverlays(@NonNull JSONArray encodedSystemUiOverlay) throws JSONException {
+    List<SystemUiOverlay> overlays = new ArrayList<>();
+    for (int i = 0; i < encodedSystemUiOverlay.length(); ++i) {
```

Comment:
Same subjective `for : each` note here.

Reply:
Same issue as above, foreach does not appear to be supported for JSONArray.